arisia-remote/app/arisia/controllers/LoginController.scala

```diff
@@ -28,15 +32,28 @@ class LoginController (
 
   def login(): EssentialAction = Action.async(controllerComponents.parsers.tolerantJson[LoginRequest]) { implicit request =>
     val req = request.body
-    loginService.checkLogin(req.id, req.password).map {
+    loginService.checkLogin(req.id, req.password).flatMap {
       _ match {
         case Some(user) => {
-          val json = Json.toJson(user)
-          val jsStr = Json.stringify(json)
-          Ok(jsStr).withSession(userKey -> jsStr).as(JSON)
+          loginService.getPermissions(user.id).map { permissions =>
+            val allowed =
+              if (earlyAccessOnly)
+              // We're in early-access mode, so the general public is *not* allowed in:
```

Comment:
Heh -- you have just stumbled in the *the* most towering arguments in the Scala community today.  Suffice it to say, Scala 3 is moving in the opposite direction, of making brackets always optional and making indentation significant, a la Python.  (I hate this with a burning passion, and have been engaged in a year-long argument about it with the head of the Scala project.)

I'm on the fence about always requiring brackets, but it's not a terrible idea.  Really, the main problem here is that I haven't had time to set up scalafmt and pre-commit, to enforce the style rules automatically.

Reply:
In case you're morbidly curious about the towering argument, here is the current incarnation, still running:

https://contributors.scala-lang.org/t/feedback-sought-optional-braces/4702